code/game/machinery/vending/vendor_types/medical.dm

```diff
@@ -222,63 +450,182 @@
-		list("Pill Bottle (Dylovene)", round(scale * 2), /obj/item/storage/pill_bottle/antitox, VENDOR_ITEM_REGULAR),
-		list("Pill Bottle (Inaprovaline)", round(scale * 2), /obj/item/storage/pill_bottle/inaprovaline, VENDOR_ITEM_REGULAR),
-		list("Pill Bottle (Kelotane)", round(scale * 2), /obj/item/storage/pill_bottle/kelotane, VENDOR_ITEM_REGULAR),
-		list("Pill Bottle (Peridaxon)", round(scale * 1), /obj/item/storage/pill_bottle/peridaxon, VENDOR_ITEM_REGULAR),
-		list("Pill Bottle (Tramadol)", round(scale * 2), /obj/item/storage/pill_bottle/tramadol, VENDOR_ITEM_REGULAR),
+		list("Pill Bottle (Bicaridine)", round(scale * 4), /obj/item/storage/pill_bottle/bicaridine, VENDOR_ITEM_REGULAR),
+		list("Pill Bottle (Dexalin)", round(scale * 4), /obj/item/storage/pill_bottle/dexalin, VENDOR_ITEM_REGULAR),
+		list("Pill Bottle (Dylovene)", round(scale * 4), /obj/item/storage/pill_bottle/antitox, VENDOR_ITEM_REGULAR),
+		list("Pill Bottle (Inaprovaline)", round(scale * 4), /obj/item/storage/pill_bottle/inaprovaline, VENDOR_ITEM_REGULAR),
+		list("Pill Bottle (Kelotane)", round(scale * 4), /obj/item/storage/pill_bottle/kelotane, VENDOR_ITEM_REGULAR),
+		list("Pill Bottle (Peridaxon)", round(scale * 3), /obj/item/storage/pill_bottle/peridaxon, VENDOR_ITEM_REGULAR),
+		list("Pill Bottle (Tramadol)", round(scale * 4), /obj/item/storage/pill_bottle/tramadol, VENDOR_ITEM_REGULAR),
 
 		list("MEDICAL UTILITIES", -1, null, null),
 		list("Emergency Defibrillator", round(scale * 3), /obj/item/device/defibrillator, VENDOR_ITEM_REGULAR),
 		list("Surgical Line", round(scale * 2), /obj/item/tool/surgery/surgical_line, VENDOR_ITEM_REGULAR),
 		list("Synth-Graft", round(scale * 2), /obj/item/tool/surgery/synthgraft, VENDOR_ITEM_REGULAR),
-		list("Hypospray", round(scale * 2), /obj/item/reagent_container/hypospray/tricordrazine, VENDOR_ITEM_REGULAR),
+		list("Hypospray", round(scale * 3), /obj/item/reagent_container/hypospray/tricordrazine, VENDOR_ITEM_REGULAR),
 		list("Health Analyzer", round(scale * 5), /obj/item/device/healthanalyzer, VENDOR_ITEM_REGULAR),
 		list("M276 Pattern Medical Storage Rig", round(scale * 2), /obj/item/storage/belt/medical, VENDOR_ITEM_REGULAR),
 		list("Medical HUD Glasses", round(scale * 3), /obj/item/clothing/glasses/hud/health, VENDOR_ITEM_REGULAR),
-		list("Stasis Bag", round(scale * 2), /obj/item/bodybag/cryobag, VENDOR_ITEM_REGULAR),
+		list("Stasis Bag", round(scale * 3), /obj/item/bodybag/cryobag, VENDOR_ITEM_REGULAR),
 		list("Syringe", round(scale * 7), /obj/item/reagent_container/syringe, VENDOR_ITEM_REGULAR)
 	)
 
+/obj/structure/machinery/cm_vending/sorted/medical/populate_product_list_and_boxes(scale)
+	. = ..()
+
+	// If this is groundside and isn't dynamically changing we will spawn with stock randomly removed from it
+	if(vend_flags & VEND_STOCK_DYNAMIC)
+		return
+	if(Check_WO())
+		return
+	var/turf/location = get_turf(src)
+	if(location && is_ground_level(location.z))
+		random_unstock()
```

Comment:
this has proven a bit tricky in that sometimes survs roll for bust on defibs in all the med vendors or roll for bust on belts or roll for bust on pills. rare but it's happened a few times and never without comment when it does happen.

my question is whether there'd be a way to create a shared pool of contents randomly split between the vendors, but that way you might always guarantee one defib, one bottle of bicaridine, etc. (discretion at what's necessary here to guarantee also), but it's not guaranteed all in one place?

i assume that implementation would look very different too and probably require a vendor subtype?

Reply:
If you want it overall, it's a simple probability/statistical problem. They would just need to adjust the probabilities in the unstocking.

Reply:
I do not think it necessary to guarantee survivors can find a defib in a vendor.

Reply:
> I do not think it necessary to guarantee survivors can find a defib in a vendor.

Some maps have not enought vendors , also the "doctor" survs deserves at least a basic loadout since thats its specialty

Reply:
> > I do not think it necessary to guarantee survivors can find a defib in a vendor.
> 
> Some maps have not enought vendors( lv only has one or two i think , while solaris and others can have up to 6) , also the "doctor" survs deserves at least a basic loadout since thats its specialty